include/boost/geometry/strategies/geographic/area_geographic.hpp

```diff
@@ -0,0 +1,212 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fisikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_AREA_GEOGRAPHIC_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_AREA_GEOGRAPHIC_HPP
+
+#include <boost/geometry/formulas/area_formulas.hpp>
+#include <boost/geometry/formulas/thomas_inverse.hpp>
+//#include <boost/math/special_functions/atanh.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace area
+{
+
+/*!
+\brief Geographic area calculation by trapezoidal rule plus integral
+approximation that gives the ellipsoidal correction
+
+}
+
+*/
+
+template
+<
+    typename PointOfSegment,
+    typename Strategy = void,
+    const std::size_t SeriesOrder = 2,
+    bool ExpandEpsN = true,
+    bool LongSegment = false,
+    typename Spheroid = void,
+    typename CalculationType = void
+>
+class area_geographic
+{
+
+    //Select default types for Strategy, Spheroid and CalculationType
+    //in case they are not set
+
+    typedef typename boost::mpl::if_c
+    <
+        boost::is_void<CalculationType>::type::value,
+        typename select_most_precise
+            <
+                typename coordinate_type<PointOfSegment>::type,
+                double
+            >::type,
+        CalculationType
+    >::type CT;
+
+    typedef typename boost::mpl::if_c
+    <
+        boost::is_void<Strategy>::type::value,
+        typename geometry::formula::thomas_inverse
+            <
+                CT,
+                false,
+                true,
+                true
+            >,
+        Strategy
+    >::type AzimuthStrategy;
```

Comment:
So if template was passed above instead of a type like this:

    template <typename, bool, bool, bool, bool ,bool> class InverseFormula = geometry::formula::thomas_inverse

this would look like this:

   typedef InverseFormula<CT, false, true, true> azimuth_formula;

If we decided to get rid of `InverseFormula`/`Policy` then here a formula type would be instantiated based on the `Order` or `Tag`.